src/Parser/Shaman/Restoration/Modules/Talents/CrashingWaves.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+import SpellLink from 'common/SpellLink';
+import SPELLS from 'common/SPELLS';
+import { formatPercentage } from 'common/format';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import HIT_TYPES from 'Parser/Core/HIT_TYPES';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import StatTracker from 'Parser/Core/Modules/StatTracker';
+import calculateEffectiveHealing from 'Parser/Core/calculateEffectiveHealing';
+
+const BUFFER_WAVE = 750;
+const BUFFER_SURGE = 100;
+const crashingWavesIncrease = 0.1;
+
+class CrashingWaves extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    statTracker: StatTracker,
+  };
+  healing = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasTalent(SPELLS.CRASHING_WAVES_TALENT.id);
+  }
+
+  on_byPlayer_heal(event) {
+    const spellId = event.ability.guid;
+
+    if (spellId === SPELLS.HEALING_SURGE_RESTORATION.id) {
+
+      const hasTidalWaves = this.combatants.selected.hasBuff(SPELLS.TIDAL_WAVES_BUFF.id, event.timestamp, BUFFER_SURGE, BUFFER_SURGE);
+      const isCrit = event.hitType === HIT_TYPES.CRIT;
+      if(!hasTidalWaves || !isCrit) {
+        return;
+      }
+      const currentCritPercent = this.statTracker.currentCritPercentage;
+      const crashingWavesContribution = crashingWavesIncrease / (crashingWavesIncrease + currentCritPercent);
+      this.healing += calculateEffectiveHealing(event, crashingWavesContribution);
+
+    } else if (spellId === SPELLS.HEALING_WAVE.id) {
+
+      const hasTidalWaves = this.combatants.selected.hasBuff(SPELLS.TIDAL_WAVES_BUFF.id, event.timestamp, BUFFER_WAVE, BUFFER_WAVE);
+      if(!hasTidalWaves) {
+        return;
+      }
+      this.healing += calculateEffectiveHealing(event, crashingWavesIncrease);
```

Comment:
Same as surge, remember base increase from tidal waves (assuming it's additive with the base tital wave increase). I assume it is intentional that you just consider the haste increase a direct healing increase? It will slightly overvalue the talent since it doesn't account for mana & more tidal wave buffs required, but that probably too complex so I understand if you left it out.